src/reactor.cc

```diff
@@ -0,0 +1,375 @@
+#include "reactor.h"
+
+#include "boost/shared_ptr.hpp"
+
+namespace tricycle {
+
+Reactor::Reactor(cyclus::Context* ctx) : cyclus::Facility(ctx) {
+  // capacities set somewhat arbitrarily
+  fuel_tracker.Init({&tritium_storage}, 1000.0);
+  blanket_tracker.Init({&blanket}, 100000.0);
+}
+
+std::string Reactor::str() {
+  return Facility::str();
+}
+
+void Reactor::Tick() {
+  if (sufficient_tritium_for_operation) {
+    SequesterTritium();
+    OperateReactor(TBR);
+    blanket_fill_policy.Start();
+    Reactor::RecordStatus("Online", fusion_power);
+  } else {
+    Reactor::RecordStatus("Shut-down", 0);
+  }
+
+
+  DecayInventory(tritium_storage);
+  DecayInventory(tritium_excess);
+  DecayInventory(tritium_sequestered);
+
+  ExtractHelium(tritium_storage);
+  ExtractHelium(tritium_excess);
+
+  if (!tritium_storage.empty() && sufficient_tritium_for_operation) {
+    double surplus = std::max(
+      tritium_storage.quantity() - reserve_inventory, 0.0);
+
+    cyclus::Material::Ptr storage_fuel = tritium_storage.Pop();
+    tritium_excess.Push(storage_fuel->ExtractQty(surplus));
+
+    if (surplus > 0.0) {
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from storage to excess");
+    }
+    tritium_storage.Push(storage_fuel);
```

Comment:
You should be able to `Pop(surplus)` and avoid having to extract and push back onto the buffer.

Also only necessary if `surplus > 0`

```suggestion

    if (surplus > 0.0) {

      tritium_excess.Push(tritium_storage.Pop(surplus));

      RecordOperationalInfo(
          "Tritium Moved",
          std::to_string(surplus) + "kg of T moved from storage to excess");
    }
```

Reply:
I most certainly can! I did this manually since I also incorporated the suggestion below. Good catch!